### PR TITLE
[CARBONDATA-2361] Refactoring ReadCommittedStatus

### PR DESCRIPTION
**Pull request by sounakr**
Refactor ReadCommittedStatus.

 - [ ] Any interfaces changed?
 
 - [ ] Any backward compatibility impacted?
 
 - [ ] Document update required?

 - [ ] Testing done
        Please provide details on 
        - Whether new unit test cases have been added or why no new tests are required?
        - How it is tested? Please attach test report.
        - Is it a performance related change? Please attach the performance test report.
        - Any additional information to help reviewers in testing this change.
       
 - [ ] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA.

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/3898/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/5119/

**Comment by sounakr** [comment]
Retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/5121/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/3900/

File: core/src/main/java/org/apache/carbondata/core/datamap/Segment.java [hunk 1]
```diff
@@ -36,9 +39,23 @@
 
   private String segmentFileName;
 
+  private ReadCommittedScope readCommittedScope;
```
Comment (jackylk) [inline core/src/main/java/org/apache/carbondata/core/datamap/Segment.java]: please provide comment for this variable

File: core/src/main/java/org/apache/carbondata/core/datamap/Segment.java [hunk 2]
```diff
@@ -80,6 +115,22 @@ public static Segment toSegment(String segmentId) {
     return new Segment(segmentId, null);
   }
 
+  /**
+   * SegmentId can be combination of segmentNo and segmentFileName
+   * @param segmentId
```
Comment (jackylk) [inline core/src/main/java/org/apache/carbondata/core/datamap/Segment.java]: provide comment or remove it

File: core/src/main/java/org/apache/carbondata/core/datamap/Segment.java [hunk 3]
```diff
@@ -80,6 +115,22 @@ public static Segment toSegment(String segmentId) {
     return new Segment(segmentId, null);
   }
 
+  /**
+   * SegmentId can be combination of segmentNo and segmentFileName
```
Comment (jackylk) [inline core/src/main/java/org/apache/carbondata/core/datamap/Segment.java]: can you describe more what is in the SegmentId

File: core/src/main/java/org/apache/carbondata/core/datamap/Segment.java [hunk 4]
```diff
@@ -36,9 +39,23 @@
...
+  }
+
+  public Segment(String segmentNo, String segmentFileName, ReadCommittedScope readCommittedScope) {
+    this.segmentNo = segmentNo;
+    this.segmentFileName = segmentFileName;
+    this.readCommittedScope = readCommittedScope;
+  }
+
+
+  public Map<String, String> getCommittedIndexFile() throws IOException {
```
Comment (jackylk) [inline core/src/main/java/org/apache/carbondata/core/datamap/Segment.java]: add comment for this function, what is the return type (what map to what)

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/5152/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/3936/

**Comment by sounakr** [comment]
Retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/4026/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/5221/

File: core/src/main/java/org/apache/carbondata/core/datamap/Segment.java (hunk 1 above)
Comment (sounakr) [reply core/src/main/java/org/apache/carbondata/core/datamap/Segment.java]: Done

File: core/src/main/java/org/apache/carbondata/core/datamap/Segment.java (hunk 2 above)
Comment (sounakr) [reply core/src/main/java/org/apache/carbondata/core/datamap/Segment.java]: Done

File: core/src/main/java/org/apache/carbondata/core/datamap/Segment.java (hunk 3 above)
Comment (sounakr) [reply core/src/main/java/org/apache/carbondata/core/datamap/Segment.java]: Done

File: core/src/main/java/org/apache/carbondata/core/datamap/Segment.java (hunk 4 above)
Comment (sounakr) [reply core/src/main/java/org/apache/carbondata/core/datamap/Segment.java]: Done

**Comment by sounakr** [comment]
Retest this please.

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/4064/

**Comment by jackylk** [comment]
LGTM

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/5246/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/4070/

**Pull request closed without merging**